### PR TITLE
Add support to delete vm from inventory. 

### PR DESCRIPTION
**Pull request by pgbidkar**
##### SUMMARY
Fix issue https://github.com/ansible/ansible/issues/55572
Add support to delete VM from inventory.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
vmware_guest

##### ADDITIONAL INFORMATION
```
ansible 2.10.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/vmware/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vmware/Ansible/ansible/lib/ansible
  executable location = /home/vmware/Ansible/ansible/bin/ansible
  python version = 3.5.2 (default, Nov 17 2016, 17:05:23) [GCC 5.4.0 20160609]


```

### CODE REVIEW

**Comment by sky-joker** [comment]
Hi, @pgbidkar

The source code of the vmware_guest module is bloated and complicated.
As a suggestion, I think it is better to implement the feature in a separate module.
For example, the following module.

https://github.com/ansible/ansible/pull/54647

**Comment by pgbidkar** [comment]
> Hi, @pgbidkar
> 
> The source code of the `vmware_guest` module is bloated and complicated.
> As a suggestion, I think it is better to implement the feature in a separate module.
> For example, the following module: #54647
> 
> thanks

Hi @sky-joker , Yes! I agree. Actually I thought about initially. But VM delete functionality is in vmware_guest module. To provide unregister functionality, delete also should be pulled out of vmware_guest.  If you agree, we can do like this. implement only register VM functionality in your PR https://github.com/ansible/ansible/pull/54647. I will create new module for delete and unregister functionality? Please let me know your thoughts?

Thanks for the feedback

**Comment by sky-joker** [comment]
I think it is simpler to separate the modules depending on whether to remove the VM from the disk or unregister it from the inventory :)
For example, the roles are divided as follows:

* `vmware_guest` delete components from disk
* `vmware_guest_register_operation` only performs inventory operations (register / unregister) without components being removed from disk

If implement `unregister` with vmware_guest, I think you also need to implement `register(RegisterVM_Task)`.
This can meet the requirements when you want to move VMs between ESXi in environments where vCenter is not present.
For example, if you use vmware_guest_register_operation, you can write a playbook as follows.

```yaml
---
- name: Register/UnRegister VM between ESXi
  hosts: localhost
  gather_facts: no
  vars:
    esxi1: esxi1.local
    esxi2: esxi2.local
    esxi_user: root
    esxi_password: password
    vm_name: example
  tasks:
    - name: "UnRegister {{ vm_name }} from esxi1."
      vmware_guest_register_operation:
        hostname: "{{ esxi1 }}"
        username: "{{ esxi_user }}"
        password: "{{ esxi_password }}"
        validate_certs: no
        folder: /vm
        esxi_hostname: "{{ esxi1 }}"
        name: "{{ vm_name }}"
        state: absent

    - name: "Register {{ vm_name }} to esx2."
      vmware_guest_register_operation:
        hostname: "{{ esxi2 }}"
        username: "{{ esxi_user }}"
        password: "{{ esxi_password }}"
        validate_certs: no
        folder: /vm
        esxi_hostname: "{{ esxi2 }}"
        name: "{{ vm_name }}"
        path: "[VM2] {{ vm_name }}/{{ vm_name }}.vmx"
        state: present
```

thanks

**Comment by pgbidkar** [comment]
Thanks @sky-joker for detailed info.

I believe my take is different on this. I see that, deleteVM and unregisterVM need to be in same module and registerVM itself is single module. 
If I am not wrong, in customer scenario, unregister and register VM is not done in single playbook.  

Thanks

File: test/integration/targets/vmware_guest/tasks/remove_vm_from_inventory.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Copyright: (c) 2019, Pavan Bidkar <pbidkar@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Create VM to unregister
+  vmware_guest:
+    validate_certs: False
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    name: vm_to_unregister
```
Comment (goneri) [inline test/integration/targets/vmware_guest/tasks/remove_vm_from_inventory.yml]: Please, use one of te standard VM names or the new resource will never be cleaned up, See: https://docs.ansible.com/ansible/devel/dev_guide/platforms/vmware_guidelines.html#vm-names-should-be-predictable

**Review by goneri — COMMENTED** [review]
Thank for the patch @pgbidkar, looks mostly good. Just a little comment.

File: test/integration/targets/vmware_guest/tasks/remove_vm_from_inventory.yml (hunk 1 above)
Comment (pgbidkar) [reply test/integration/targets/vmware_guest/tasks/remove_vm_from_inventory.yml]: Done.
Actually I thought test_vm1, test_vm2, test_vm3 are existing VMs in the setup. So used different name.
Done the changes now

**Review by goneri — APPROVED** [review]

File: test/integration/targets/vmware_guest/defaults/main.yml [hunk 2]
```diff
@@ -29,4 +29,4 @@ vmware_guest_test_playbooks:
   - poweroff_d1_c1_f1.yml
 #  - template_d1_c1_f0.yml
   - vapp_d1_c1_f0.yml
-  - reconfig_vm_to_latest_version.yml
```
Comment (Akasurde) [inline test/integration/targets/vmware_guest/defaults/main.yml:32]: No need to remove existing test case.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: test/integration/targets/vmware_guest/defaults/main.yml (hunk 2 above)
Comment (pgbidkar) [reply test/integration/targets/vmware_guest/defaults/main.yml:32]: Oops! Sorry by mistake I removed the testcase. Thought it is in progress PR Testcase
Added it back

**Review by Akasurde — APPROVED** [review]

**Pull request merged**